build.sbt

```diff
@@ -32,21 +32,21 @@ lazy val core: Project = project
 lazy val azure: Project = project
   .in(file("modules/azure"))
   .settings(BuildSettings.azureSettings)
-  .settings(libraryDependencies ++= Dependencies.azureDependencies)
+  .settings(libraryDependencies ++= Dependencies.azureDependencies35 ++ Dependencies.spark35RuntimeDependencies)
```

Comment:
It looks like `azureDependencies35` already brings `sparkCore35`. Why do then also have to provide `spark35RuntimeDependencies` with another `sparkCode35`?

I think I don't understand why now `${cloud}Dependencies` contains spark core dependency. Do we have to do that?

Reply:
The azure app needs to [import from hadoop at compile time](https://github.com/snowplow-incubator/snowplow-lake-loader/blob/0.4.0-rc1/modules/azure/src/main/scala/com.snowplowanalytics.snowplow.lakes/AzureTokenProvider.scala#L18-L19).  And the easiest way to do that was to add Spark as a compile time dependency.  I suppose I could have added hadoop to `azureDependencies` instead of adding spark.  Hmmm maybe that's better.

Similarly the aws app needs to [import from hadoop at compile time](https://github.com/snowplow-incubator/snowplow-lake-loader/blob/0.4.0-rc1/modules/aws/src/main/scala/com.snowplowanalytics.snowplow.lakes/AssumedRoleCredentialsProvider.scala#L18-L19).

It's also in `spark35RuntimeDependencies` because that's the only way it gets provided to the gcp app.  The gcp module does not import from spark or hadoop at compile time.

I suppose I could have done this differently.  Either....

1. I could have added hadoop to `azureDependencies` and `awsDependencies` instead of adding all of spark. This would have made it more clear what is needed at compile time.
2. Or, I could have added spark to `gcpDependencies` so at least all clouds look more similar.

Hmmmm maybe 1 is nicer.  What do you think?

Reply:
I'd go with 1. And I think `awsDependencies` already declares `hadoopAws` (similar for azure and `hadoopAzure`) dependency for compile, so we have everything we need with hadoop for each cloud, right?